lib/cadet/assessments/assessments.ex

```diff
@@ -753,6 +809,69 @@ defmodule Cadet.Assessments do
     |> Repo.transaction()
   end
 
+  def update_contest_voting_submission_status(submission = %Submission{}, question = %Question{}) do
+    count_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> Repo.all()
+      )
+
+    count_not_nil_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> where([c], not is_nil(c.score))
+        |> Repo.all()
+      )
+
+    if count_entries == count_not_nil_entries do
+      submission |> Submission.changeset(%{status: :attempted}) |> Repo.update()
+    end
+  end
+
+  defp load_contest_voting_entries(questions, assessment_id, user_id) do
+    Enum.map(
+      questions,
+      fn q ->
+        if q.type == :voting do
+          question =
+            Question
+            |> where(assessment_id: ^assessment_id)
+            |> Repo.one()
+
+          voting_question =
+            case all_submission_votes_by_question_id_and_user_id(question.id, user_id) do
+              {:ok, submission_votes} -> Map.put(q.question, :contestEntries, submission_votes)
+            end
+
+          Map.put(q, :question, voting_question)
+        else
+          q
+        end
+      end
+    )
+  end
+
+  @doc """
+  Function returning contest submission entries assigned to a user to vote for.
```

Comment:
This should be phrased "Returns contest submission entries ..."

(I know other docstrings are phrased badly, but let's not add more bad stuff)